packages/protocol/src/SavingsContract.ts

```diff
@@ -33,24 +33,32 @@ export function getOrCreateSavingsContract(
   if (version == 2) {
     let tokenEntity = token.getOrCreate(addr)
     savingsContractEntity.token = tokenEntity.id
-  }
-
-  if (massetAddress != null) {
-    savingsContractEntity.masset = massetAddress.toHexString()
-  } else {
-    let underlying = contract.try_underlying()
-    if (underlying.reverted) {
+    let fraction = contract.try_fraction()
+    if (fraction.reverted) {
```

Comment:
Also the error message below doesn't relate to this function so it would be misleading